syntax = "proto3";

package finazon;

//Stocks

message TickerStocksInfo {
    string ticker = 1;
    string country = 2;
    string currency = 3;
    // string mic = 4;
    string publisher = 4;
    string publisher_name = 5;
}

message FindTickersStocksRequest {
    string ticker = 1;
    string publisher = 2;
    string country = 3;
    string mic = 4;
    int32  page = 5;
    int32  page_size = 6;
    string cqs = 11;
    string cik = 12;
    string cusip = 13;
    string isin = 14;
    string composite_figi = 15;
    string share_figi = 16;
    string lei = 17;
    string dataset = 18;
}

message FindTickersStocksResponse {
    repeated TickerStocksInfo result = 1;
}

//Crypto

message TickerCryptoInfo {
    string ticker = 1;
    string publisher = 2;
    string publisher_name = 3;
}

message FindTickersCryptoRequest {
    string ticker = 1;
    string publisher = 2;
    string market = 3;
    int32  page = 4;
    int32  page_size = 5;
}

message FindTickersCryptoResponse {
    repeated TickerCryptoInfo result = 1;
}

//Forex

message TickerForexInfo {
    string ticker = 1;
}

message FindTickersForexRequest {
    string ticker = 1;
    int32  page = 2;
    int32  page_size = 3;
}

message FindTickersForexResponse {
    repeated TickerForexInfo result = 1;
}

//US tickers
message TickerUSInfo {
    string ticker = 1;
    string currency = 2;
    string security = 3;
    string mic = 4;
    string asset_type = 5;
    string cik = 7;
    string composite_figi = 10;
    string share_figi = 11;
    string lei = 12;
}

message FindTickersUSRequest {
    string ticker = 1;
    string mic = 2;
    int32  page = 3;
    int32  page_size = 4;
    string cqs = 5;
    string cik = 6;
    string cusip = 7;
    string isin = 8;
    string composite_figi = 9;
    string share_figi = 10;
    string lei = 11;
}

message FindTickersUSResponse {
    repeated TickerUSInfo result = 1;
}

/**
 * TickersService provide access to tickers
 */
service TickersService {
    // This API call returns an array of stocks tickers available at Finazon Data API. This list is updated daily.
    rpc FindTickersStocks(FindTickersStocksRequest) returns (FindTickersStocksResponse) {};

    // This API call returns an array of crypto tickers available at Finazon Data API. This list is updated daily.
    rpc FindTickersCrypto(FindTickersCryptoRequest) returns (FindTickersCryptoResponse) {};

    // This API call returns an array of forex tickers available at Finazon Data API. This list is updated daily.
    rpc FindTickersForex(FindTickersForexRequest) returns (FindTickersForexResponse) {};

    // This API call returns an array of US tickers available at Finazon Data API. This list is updated daily.
    rpc FindTickersUS(FindTickersUSRequest) returns (FindTickersUSResponse) {};
}